src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/model/sar/SARConditionFormatter.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.hmppshdcapi.model.sar
+
+import uk.gov.justice.digital.hmpps.hmppshdcapi.licences.conditions.ConditionMetadata
+import uk.gov.justice.digital.hmpps.hmppshdcapi.licences.conditions.V1_CONDITIONS
+import uk.gov.justice.digital.hmpps.hmppshdcapi.licences.conditions.V2_CONDITIONS
+
+typealias ConditionPolicies = Map<Int, List<ConditionMetadata>>
+
+object SARConditionFormatter {
+
+  val policyVersions = mapOf(1 to V1_CONDITIONS, 2 to V2_CONDITIONS)
+  private val conditionToPolicyVersion =
+    policyVersions.entries.flatMap { it.value }.groupingBy { it.id }.eachCount().filter { it.value == 1 }
+      .map { it.key to policyVersions.findVersion(it.key) }.toMap()
+
+  fun ConditionPolicies.getConditionText(conditionVersion: Int?, id: String) =
+    policyVersions[conditionVersion]?.find { it.id == id }?.text ?: id
+
+  fun ConditionPolicies.getConditionText(id: String) =
+    policyVersions.values.flatMap { it.filter { condition -> condition.id == id } }.takeIf { it.size == 1 }
+      ?.firstOrNull()?.text ?: id
+
+  fun getConditionText(conditionVersion: Int?, code: String): String =
+    conditionVersion?.let { policyVersions.getConditionText(it, code) } ?: policyVersions.getConditionText(code)
+
+  fun getConditionValues(values: Map<String, Any>): Map<String, Any>? =
+    values.takeIf { it.isNotEmpty() }?.mapKeys { it.key.camelToSentenceCase() }
+
+  fun ConditionPolicies.findVersion(id: String) =
+    entries.find { (_, conditions) -> conditions.any { it.id == id } }?.key
+
+  fun ConditionPolicies.getPolicyVersion(conditionIds: Collection<String>) =
+    conditionIds.map { conditionToPolicyVersion[it] }.distinct().single()
```

Comment:
try to guess what policy version it is by looking at condition ids on the licence and checking to see if any uniquely appear on one version of a policy. If so that's the policy we want.

It ignores conditions that appear on more than one policy version
It blows up if there are conditions from different policy versions on the same licence